File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l ,n = 0, len(height)
        r = n-1
        maxx = 0
        while(l<r):
            w = r-l
            h = min(height[l], height[r])
            cur = w*h
            maxx = max(cur, maxx)
            if height[l] < height[r]:
                l+=1
            else:
                r-=1
        return maxx

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0 ,n = height.size() ,r = n-1, maxx= INT_MIN;
        while(l<r){
            int w = r - l;
            int h = min(height[l], height[r]);
            int curr = w * h;
            maxx = max(maxx, curr) ;
            if(height[l] < height[r]){
                l++;
            }
            else{
                r--;
            }

        }
        return maxx;
    }
};